### PR TITLE
41 add data residency migrations module

### PR DESCRIPTION
**Pull request by YabMas**
Also:
- added http module
- start of google-cloud wrapper module. For this task only a single function was added, but it might be worth it to collect wrappers for all services we use.
- formatting for all modules
- solves issue with new restriction from clojars by automatically adding mit license to all pom's

### CODE REVIEW

**Review by mitchelkuijpers — CHANGES_REQUESTED** [review]

File: modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                             :migrationCallback)]
+        (http/request
+          {::http/base-url callback-url
+           ::http/method "POST"
+           ::http/headers
+           {"Content-Type" "application/json"
+            "Authorization" (str "Bearer " token)}
+           ::http/body {:phase phase}})))))
+
+(defn ^:private migration-handler [{:keys [installation]
```
[the remark this answers is not in the document]
Comment (mitchelkuijpers) [reply modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs]: Please chang ethis for all defn definitions

**Review by Demivk — APPROVED** [review]

File: modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs (hunk 1 above)
[the remark this answers is not in the document]
Comment (YabMas) [reply modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs]: I'm aware of this but honestly prefer ^:private as it's way more clear. I think adding defn- was a mistake and is regarded as such by the core-team, why else is there no def- defmethod- etc.

[the remark this answers is not in the document]
Comment (YabMas) [reply modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs]: But I also think it's not interesting enough to have lengthy discussions about :)

File: modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  (:require
+    [com.avisi-apps.gaps.log :as log]
+    [com.avisi-apps.gaps.http.core :as http]
+    [com.avisi-apps.gaps.google-cloud.pubsub :as pubsub]
+    [clojure.string :as str]
+    [cljs-bean.core :refer [->clj]]
+    [promesa.core :as p]
+    [camel-snake-kebab.core :as csk]))
+
+(defn ^:private get-project-id [] (.-env.GCLOUD_PROJECT js/process))
```
[the remark this answers is not in the document]
Comment (YabMas) [reply modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs:11]: Hmm, yeah maybe this is not the optimal way. But I would indeed leave it for now and see how the module evolves first. This fn is only called in one place at the moment. I also don't think we should move the burden of providing this onto the calling app, as we'll be implementing exactly this in all of them.

File: modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mitchelkuijpers) [reply modules/data-residency-migrations/src/com/avisi_apps/gaps/data_residency_migrations/core.cljs]: I agree totally fine.

**Review by mitchelkuijpers — APPROVED** [review]

**Pull request merged**